pennylane/vqe/vqe.py

```diff
@@ -528,6 +528,11 @@ def cost_fn(*qnode_args, **qnode_kwargs):
             self.cost_fn = qml.dot(coeffs, self.qnodes)
 
     def __call__(self, *args, **kwargs):
+        coeffs = self.hamiltonian.coeffs
+        if all(c == 0 for c in coeffs) or not coeffs:
+            # We have a zero Hamiltonian whose measurement statistic is zero
+            return np.array(0)
```

Comment:
Since the Hamiltonian doesn't change between evaluations, should this instead be in `__init__`? That will reduce calling overhead.